misk/src/main/kotlin/misk/environment/DeploymentModule.kt

```diff
@@ -0,0 +1,14 @@
+package misk.environment
+
+import misk.inject.KAbstractModule
+
+/** Binds [Deployment] and [Env] to make them available to services and actions */
+class DeploymentModule(
```

Comment:
After landing this PR, my plan is to:
1. Update apps to instead this in addition to `EnvironmentModule`, so that all are available during transition period.
2. Update misk and skim components to rely on `Deployment` and raw env instead of `Environment`
3. Remove uses of `EnvironmentModule` and `Environment` in misk, skim, and app code.
4. Delete `EnvironmentModule` and `Environment`.

Reply:
Plan sounds thorough and backwards compatible during the migration period 👍